java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -826,10 +827,117 @@ open class KotlinFileExtractor(
+                                        extractVariableAccess(tw.getLabelFor<DbParam>(getValueParameterLabel(id, paramIdx + paramIdxOffset)), param.type, locId, assignId, 0, id, exprStmtId)
+                                        extractExpressionExpr(defaultVal.expression, id, assignId, 1, exprStmtId)
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    // Now call the real function:
+                    val realFunctionId = useFunction<DbCallable>(f)
+                    if (f is IrConstructor) {
+                       tw.getFreshIdLabel<DbConstructorinvocationstmt>().also { thisCallId ->
+                           tw.writeStmts_constructorinvocationstmt(thisCallId, blockId, nextStmt++, id)
+                           tw.writeHasLocation(thisCallId, locId)
+                           f.valueParameters.forEachIndexed { idx, param ->
+                               extractVariableAccess(tw.getLabelFor<DbParam>(getValueParameterLabel(id, idx)), param.type, locId, thisCallId, idx, id, thisCallId)
+                           }
+                           tw.writeCallableBinding(thisCallId, realFunctionId)
+                       }
+                    } else {
+                        tw.getFreshIdLabel<DbReturnstmt>().also { returnId ->
+                            tw.writeStmts_returnstmt(returnId, blockId, nextStmt++, id)
+                            tw.writeHasLocation(returnId, locId)
+                            extractMethodAccessWithoutArgs(f.returnType, locId, id, returnId, 0, returnId, realFunctionId).also { thisCallId ->
+                                val realFnIdxOffset = if (f.extensionReceiverParameter != null) 1 else 0
+                                val paramMappings = f.valueParameters.mapIndexed { idx, param -> Triple(param.type, idx + paramIdxOffset, idx + realFnIdxOffset) } +
+                                    listOfNotNull(
+                                        dispatchReceiver?.let { Triple(it.type, realFnIdxOffset, -1) },
+                                        extReceiver?.let { Triple(it.type, 0, 0) }
+                                    )
+                                paramMappings.forEach { (type, fromIdx, toIdx) ->
+                                    extractVariableAccess(tw.getLabelFor<DbParam>(getValueParameterLabel(id, fromIdx)), type, locId, thisCallId, toIdx, id, returnId)
+                                }
+                                if (f.shouldExtractAsStatic)
+                                    extractStaticTypeAccessQualifier(f, thisCallId, locId, id, returnId)
+                                else if (f.isLocalFunction()) {
+                                    extractNewExprForLocalFunction(getLocallyVisibleFunctionLabels(f), thisCallId, locId, id, returnId)
```

Comment:
I don't think we need to extract a `new` expression for local functions. Shouldn't this be either param 0 or a static type access instead?

Reply:
Fixed in https://github.com/github/codeql/pull/10683/commits/bec948682d01f4b87179d4b1a056b6c608ea953a